feature/experimental/p4rt/otg_tests/traceroute_packetin_test/README.md

```diff
@@ -1,34 +1,124 @@
 # P4RT-5.1: Traceroute: PacketIn
 
-
 ## Summary
 
-Verify that Traceroute packets are punted with correct metadata.
+Test FRR behaviors with encapsulation scenarios.
+
+## Topology
+
+ATE port-1 <------> port-1 DUT
+DUT port-2 <------> port-2 ATE
+DUT port-3 <------> port-3 ATE
+DUT port-4 <------> port-4 ATE
+DUT port-5 <------> port-5 ATE
+DUT port-6 <------> port-6 ATE
+DUT port-7 <------> port-7 ATE
+DUT port-8 <------> port-8 ATE
+
+## Baseline setup
+
+*   Setup equivalent to vrf_policy_driven_te, including GRibi programming.
+
+*   Install a BGP route resolved by ISIS in default VRF to route traffic out of DUT port-8 for 1.2.3.4.
+
+*   Enable the P4RT server on the device.
+
+*   Connect a P4RT client and configure the forwarding pipeline. InstallP4RT table entries required for traceroute.
 
 
 ## Procedure
 
-*	Connect ATE port-1 to DUT port-1, ATE port-2 to DUT port-2.
-*	TODO: Install the set of routes on the device.
-*	Enable the P4RT server on the device.
-*	Connect a P4RT client and configure the forwarding pipeline. InstallP4RT table 	entries required for traceroute.
-*	Send IPv4 packets from the ATE with TTL=1 and verify that packets with TTL=1 are received by the client.
-*	Send IPv6 packets from the ATE with HopLimit=1 and verify that packets with HopLimit=1 are received by the client.
-*	Verify that the packets have both ingress_singleton_port and egress_singleton_port metadata set.
+At the start of each of the following scenarios, ensure:
+
+*   All ports are up and baseline is reset as above.
+
+Unless otherwise specified, all the tests below should use traffic with
+`dscp_encap_a_1`.
+
+### Test-1
+
+Tests that traceroute for a packet coming from the cluster respects
+the encap tunnels.
+
+*   Send packets to DUT port-1 with TTL=1. The outer v4 header has the destination
+    addresses 138.0.11.8. verify that packets with TTL=1 are received by the client.
+
+*   Verify that the packets have both ingress_singleton_port and egress_singleton_port metadata set.
+egress_singletone_port should have port 2 with weight 1, port 3 weight 3, and port 4 with weight 6.
+
+
+### Test-2
+
+Tests that traceroute for a packet coming from the cluster that should
```

Comment:
describe should the packet get matched to a vrf selection policy or matched by the P4RT rules.